components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -416,6 +416,33 @@ impl<E: KvEngine> CoprocessorHost<E> {
         }
     }
 
+    pub fn post_exec(
```

Comment:
Why not use `post_apply`?

Reply:
The existing `post_apply` observer is not enough. Only `pre_exec` can observe more necessary execution information we need, at a earlier time than `post_apply`.

Reply:
Please add some comment to describe the difference.

Reply:
addressed